integration/create_integration_test.go

```diff
@@ -0,0 +1,148 @@
+)
+
+const integrationTestPath = "createIntegrationTestData/"
+
+func TestCreateIntegration(t *testing.T) {
+	fhSyncServer := &ProvisionServiceParams{
+		ServiceName: "fh-sync-server",
+		Namespace:   fmt.Sprintf("--namespace=%s", *namespace),
+		Params: []string{
+			"-p MONGODB_USER_NAME=fhsync",
+			"-p MONGODB_USER_PASSWORD=fhsyncpass",
+			"-p MONGODB_ADMIN_PASSWORD=pass",
+		},
+	}
+
+	keycloak := &ProvisionServiceParams{
+		ServiceName: "keycloak",
+		Namespace:   fmt.Sprintf("--namespace=%s", *namespace),
+		Params: []string{
+			"-p ADMIN_NAME=admin",
+			"-p ADMIN_PASSWORD=pass",
+		},
+	}
+
+	// create fh-sync-server service instance
+	createInstance(t, fhSyncServer)
+	fhSyncID := getInstanceID(t, fhSyncServer)
+
+	// create keycloak service instance
+	createInstance(t, keycloak)
+	keycloakID := getInstanceID(t, keycloak)
+
+	tests := []struct {
+		name       string
+		fhSyncID   string
+		keycloakID string
+		fixture    string
+		validate   func(t *testing.T, sb *v1beta1.ServiceBinding)
+	}{
+		{
+			name:       "missing/incorrect arguments",
+			fhSyncID:   "",
+			keycloakID: "",
+			fixture:    "missing-args.golden",
+		},
+		{
+			name:       "create integration returns ready status",
+			fhSyncID:   fhSyncID,
+			keycloakID: keycloakID,
+			validate: func(t *testing.T, sb *v1beta1.ServiceBinding) {
+				expectedType := "Ready"
+				expectedStatus := "True"
+
+				if actualType := string(sb.Status.Conditions[0].Type); actualType != expectedType {
+					t.Fatalf("Expected condition type to be '%s' but got '%s'", expectedType, actualType)
+				}
+
+				if actualStatus := string(sb.Status.Conditions[0].Status); actualStatus != expectedStatus {
+					t.Fatalf("Expected condition status to be '%s' but got '%s'", expectedStatus, actualStatus)
+				}
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// create service binding
+			args := []string{"create", "integration", test.fhSyncID, test.keycloakID, "--namespace=" + *namespace}
+			cmd := exec.Command(*executable, args...)
+			output, err := cmd.CombinedOutput()
+			if err != nil {
+				t.Fatalf("Failed to create binding: %v", err)
+			}
+			if *update {
+				WriteSnapshot(t, integrationTestPath+test.fixture, output)
+			}
+
+			if test.fixture != "" {
+				actual := string(output)
+				expected := LoadSnapshot(t, integrationTestPath+test.fixture)
+				if actual != expected {
+					t.Fatalf("actual = \n%s, expected = \n%s", actual, expected)
+				}
+			}
+
+			if test.validate != nil {
+				sb := getBinding(t)
+				test.validate(t, sb)
+			}
+		})
+	}
+}
+
+func createInstance(t *testing.T, si *ProvisionServiceParams) {
+	args := []string{"create", "serviceinstance", si.ServiceName, si.Namespace}
+	args = append(args, si.Params...)
+	cmd := exec.Command(*executable, args...)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		t.Fatalf("Failed to create service instance %s: %v", si.ServiceName, err)
```

Comment:
The `err` doesn't seem to be giving much information on what the actual error is. It seems to just be something along the lines of `exit status 1`. I suggest printing out the `output` here as it seems that this contains the actual error message

Reply:
Yeah i noticed that too. I'll give it a go, thanks!